structure/0039-pr-policy.md

```diff
@@ -0,0 +1,74 @@
+# Quilt PR Policy
+# Summary and Scope
+This RFC defines a system for merging Pull Requests (PR)s into Quilt projects. It applies to any repositories which opt-in to this system by adding a section (given below) to their CONTRIBUTING file.
+This RFC does not seek to require projects to use this process, but we encourage all projects expecting many pull requests to do so.
+
+# Motivation
+This RFC intends to make the process of creating, reviewing, and merging PRs easier for both Quilt teams and potential contributors.
+
+The goals for the PR process are:
+- **Democratic** -- Features are added through consensus of the team, and not through a benevolent dictator for life (BDFL) who has total control over the project.
+- **Quick** -- Tools should be in place to allow the team to shut down bikeshedding and say good enough on features if needed. However, this doesn't mean that pull requests should be easily forced through.
+- **Hands-off** -- Every team member shouldn't have to approve every PR. The team as a whole has a responsibility to ensure PRs are reviewed in a timely manner, but no person should have a responsibility to view every change.
+
+It is **not a goal** of this PR to be fool-proof from bad actors within the Quilt technical teams abusing the system. Being on any team in Quilt requires trust, and bad actors should be removed instead of worked around.
+
+# Explanation
+## Vocab
+- **Required Approvals**: The total number of approvals needed by the team assigned to a PR, or that team's superiors.
+    - An approval is defined as an approving review using the GitHub review feature
+- **Minimum Team Approvals**: The amount of approvals that must come directly from the assigned team. When multiple teams are assigned to a PR, this number applies to each team.
+    - This prevents PRs from getting merged by Technical Leads/etc without any oversight by the responsible team
+
+## Acceptance Criteria
+A Pull Request is considered to have been approved when all of the following are true:
+- The Pull Request has received the needed amount of Required Approvals and Minimum Team Reviews
+- The Pull Request has no changes requested reviews by any members of the team
+- Nobody in the team has had their review requested on the Pull Request using the system on Github
+    - This includes someone requesting the review of themselves or another person
+
+## Final Comment Period (FCP)
```

Comment:
Let us take a hypothetical scenario where a PR has been approved through this policy and has been merged, and then a critical issue is found in the implementation of the PR which needs immediate correcting. Let us then say that the same PR author creates a new PR immediately after the discovery of said bug in order to fix or remedy it in some way. 

Let us take that this RFC is passed as it currently stands:
 - If the PR is subjected to the same process as defined in this RFC as augmented by the repository's contributing information, then the fix for said hypothetical critical issue would be delayed for at least the time for the final comment period.
 - If a maintainer of the repository manually commits the same fix as contributed by the PR, it would essentially be an act of fiat from the maintainer to bypass the PR process and manually integrate the PR's changes, and sets the same precedent within the project.

Therefore, either the fix for the critical issue is delayed, or a precedent of maintainers being able to effectively bypass this PR policy is set outside of this RFC.

As a solution for this hypothetical scenario, I suggest a clause be added wherein if the active members of the team show unanimous consensus for the merging of a PR, with some additional condition(s) to prevent the abuse of this by the team (such as agreement by a technical lead and/or Quilt admin), then the PR may be merged bypassing the final comment period.

Reply:
I don't think we need to have a specific defined method of going around this RFC, but it should be left to the individual teams on how they wish to approach this problem.

For example, in my proposed PR policy for QSL, I add an "urgent" triage category that has no FCP for this exact use case. Additionally, I define in the policy that those with push access may bypass the policy for trivial changes like typos after double checking with the team that they think it's okay. Technically, two bad actors could merge whatever they want by abusing the `urgent` tag, (or just solo pushing whatever they want) but it's unlikely that those people would remain members of the team for very long.

Other projects may want to make different decisions about these extraneous circumstances, and that's fine! This process is more of a guideline than a strict definition.